From 8168627f48b246fae4cf40a4fac225f50880df2f Mon Sep 17 00:00:00 2001
From: Arulanbu Balusamy <quic_abalusam@quicinc.com>
Date: Wed, 9 Oct 2024 17:20:12 +0530
Subject: [PATCH] UPSTREAM:CP:KW:mac80211:ath12k:Fix KW issue to avoid banned function.

As 'sprintf' has been deprecated and it is not safe, replace it
with a safe function 'snprintf'



Patch-dependency: [None]
Patch-work:       [None]
Fixes:            [None]

Signed-off-by: Arulanbu Balusamy <quic_abalusam@quicinc.com>
---
 net/mac80211/debugfs.c        | 2 +-
 net/mac80211/debugfs_netdev.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mac80211/debugfs.c b/net/mac80211/debugfs.c
index 686e7a7..03f6b4a 100644
--- a/net/mac80211/debugfs.c
+++ b/net/mac80211/debugfs.c
@@ -650,7 +650,7 @@ static ssize_t queues_read(struct file *file, char __user *user_buf,
 
 	spin_lock_irqsave(&local->queue_stop_reason_lock, flags);
 	for (q = 0; q < local->hw.queues; q++)
-		res += sprintf(buf + res, "%02d: %#.8lx/%d\n", q,
+		res += snprintf(buf + res, sizeof(buf) + res, "%02d: %#.8lx/%d\n", q,
 				local->queue_stop_reasons[q],
 				skb_queue_len(&local->pending[q]));
 	spin_unlock_irqrestore(&local->queue_stop_reason_lock, flags);
diff --git a/net/mac80211/debugfs_netdev.c b/net/mac80211/debugfs_netdev.c
index affd09c..ca7f2c3 100644
--- a/net/mac80211/debugfs_netdev.c
+++ b/net/mac80211/debugfs_netdev.c
@@ -1050,7 +1050,7 @@ void ieee80211_debugfs_rename_netdev(struct ieee80211_sub_if_data *sdata)
 	if (IS_ERR_OR_NULL(dir))
 		return;
 
-	sprintf(buf, "netdev:%s", sdata->name);
+	snprintf(buf, sizeof(buf), "netdev:%s", sdata->name);
 	debugfs_rename(dir->d_parent, dir, dir->d_parent, buf);
 }
 
-- 
2.34.1

